Taskfile.yml

```diff
@@ -0,0 +1,93 @@
+version: '3'
+# silent: true
+output: group
+dotenv: [.env]
+
+# Default environment variables for projects
+env:
+  DOCKER_BUILDKIT: 1
+
+vars:
+  TODAY: '{{ now | date "2006-01-02T15:04:05-07:00" }}'
+  MODEL_DIRECTORY: cache/models
+  black: \033[:0;30m
+  red: \033[:0;31m
+  green: \033[:0;32m
+  orange: \033[:0;33m
+  blue: \033[:0;34m
+  purple: \033[:0;35m
+  cyan: \033[:0;36m
+  light_gray: \033[:0;37m
+  dark_gray: \033[:1;30m
+  light_red: \033[:1;31m
+  light_green: \033[:1;32m
+  yellow: \033[:1;33m
+  light_blue: \033[:1;34m
+  light_purple: \033[:1;35m
+  light_cyan: \033[:1;36m
+  white: \033[:1;37m
+  nocolor: \u001b[0m
+  reversed: \u001b[7m
+tasks:
+  default:
+    cmds:
+      - task: list
+  list:
+    desc: list tasks
+    prefix: ' '
+    cmds:
+      - task --list --color=true --list-all
+  vars:
+    desc: variable output
+    cmds:
+      - |
+        echo -e "{{.light_gray}}=== ℹ {{ .reversed }} Variable Info from Task {{ .nocolor }} === "
+        echo -e "{{.light_gray}}MODEL_DIRECTORY          {{.nocolor}}: {{ .orange}}{{ .MODEL_DIRECTORY }} {{ .nocolor }}"
+        echo -e "{{.light_gray}}DOCKER_BUILDKIT          {{.nocolor}}: {{ .orange}}{{ .DOCKER_BUILDKIT }} {{ .nocolor }}"
+        echo -e "{{.light_gray}}UI                       {{.nocolor}}: {{ .orange}}{{ .UI }} {{ .nocolor }}"
+
+  ### Core Task ###
+  init:
+    desc: Download models, build initial docker-compose setup
+    cmds:
+    - task: download
+    - task: update
+  compose:build:
+    desc: setup tooling for project and download dependencies
+    cmds:
+      - |
+        docker compose --profile download up --build
+  compose:up:
+    desc: 'compose up with the current profile of: {{ .UI }}'
+    cmds:
+    - task: vars
+    - docker compose --profile {{ .UI }} up
+  update:
+    desc: run updates
+    interactive: true
+    cmds:
+    - task: vars
+    - |
+      docker compose --profile {{ .UI }} up --build
+    - task: download
+  download:
+    desc: get extras downloaded
+    cmds:
+    - |
+      echo '👉 This might take a few minutes. However, it will resume downloading if interuppted and ran again'
```

Comment:
The download task does the resume, and I think it uses the checksum, not sure.
I opted for the test file rather than fingerprinted checksum for simplicity. 🤷🏻